scripts.py

```diff
@@ -1522,7 +1513,10 @@ def _run_migration(self, migration_path, timestamp):
 
         if migration_path.endswith('.sql'):
             with open(migration_path) as clause:
-                sql = clause.read()
+                # By wrapping the action in a transation, we can avoid
+                # rolling over errors and losing data in files
+                # with multiple interrelated SQL actions.
+                sql = 'BEGIN;\n%s\nCOMMIT;' % clause.read()
```

Comment:
This helps avoid some of the SQL problems I've had in recent migrations. This way, if it errors it errors. No more partial migrations.